### PR TITLE
memcached: Persistent connections with IO timeouts

### PR DESCRIPTION
**Pull request by rpv-tomsk**
Implemented persistent connections. It is a must-have feature for statistics daemon.

### CODE REVIEW

**Comment by rpv-tomsk** [comment]
Ping.

File: src/memcached.c [hunk 1]
```diff
@@ -124,77 +144,153 @@ static int memcached_connect_inet(memcached_t *st) {
...
-  if (fd < 0) {
+  memcached_connect(st);
+  if (st->fd < 0) {
     ERROR("memcached plugin: Instance \"%s\" could not connect to daemon.",
           st->name);
     return -1;
   }
 
-  status = (int)swrite(fd, "stats\r\n", strlen("stats\r\n"));
+  struct pollfd pollfd;
```
Comment (octo) [inline src/memcached.c]: As above: please use a struct initializer; consider a compound literal.

**Comment by rpv-tomsk** [comment]
Patch updated.

Thanks for a review!

**Comment by rpv-tomsk** [comment]
Added a small patch from #2300. That is related.

**Pull request merged**